python/pyspark/errors/exceptions/connect.py

```diff
@@ -45,14 +47,34 @@ class SparkConnectException(PySparkException):
     """
 
 
-def convert_exception(info: "ErrorInfo", message: str) -> SparkConnectException:
+def convert_exception(
+    info: "ErrorInfo", message: str, stub: "SparkConnectServiceStub"
+) -> SparkConnectException:
     classes = []
     if "classes" in info.metadata:
         classes = json.loads(info.metadata["classes"])
 
+    stack_trace = ""
+
     if "stackTrace" in info.metadata:
-        stackTrace = info.metadata["stackTrace"]
-        message += f"\n\nJVM stacktrace:\n{stackTrace}"
+        stack_trace = info.metadata["stackTrace"]
+
+    if "userId" in info.metadata and "sessionId" in info.metadata and "errorId" in info.metadata:
+        req = pb2.GetErrorInfoRequest()
+        req.user_context.user_id = info.metadata["userId"]
+        req.session_id = info.metadata["sessionId"]
+        req.error_ids.append(info.metadata["errorId"])
+
+        try:
+            result = GetErrorInfoResult.fromProto(stub.GetErrorInfo(req))
+            if result is not None and result.error_id == info.metadata["errorId"]:
+                message = result.message
+                stack_trace = result.stack_trace
+        except (Exception,):
```

Comment:
```suggestion
        except Exception:
```

Which exception do we expect to ignore here? Would be great if we can catch narrower scope.